pkg/sensors/tracing/selectors.go

```diff
@@ -230,6 +266,362 @@ func populateAddr6FilterMap(
 	return nil
 }
 
+// We need a populating function for each of the 6 string maps because each one
+// uses a different size map key and golang doesn't really like mixing types.
+func populateStringFilterMaps1(
+	k *selectors.KernelSelectorState,
+	pinPathPrefix string,
+	outerMap *ebpf.Map,
+) error {
+	maxEntries := k.StringMapsMaxEntries(1)
+	for i, am := range k.StringMaps1() {
+		nrEntries := uint32(len(am))
+		// Versions before 5.9 do not allow inner maps to have different sizes.
+		// See: https://lore.kernel.org/bpf/20200828011800.1970018-1-kafai@fb.com/
+		if !kernels.MinKernelVersion("5.9") {
+			nrEntries = uint32(maxEntries)
+		}
+		err := populateStringFilterMap1(pinPathPrefix, outerMap, uint32(i), am, nrEntries)
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func populateStringFilterMap1(
+	pinPathPrefix string,
+	outerMap *ebpf.Map,
+	innerID uint32,
+	innerData map[[selectors.StringMapsSize1]byte]struct{},
+	maxEntries uint32,
+) error {
+	innerName := fmt.Sprintf("string_maps_1_%d", innerID)
```

Comment:
Do we need 6 copies of this function? Can we just use an additional argument and a single function?

Reply:
I looked at this, but as the key to the golang maps and the BPF maps is different for each map, I came to the conclusion that the only way in golang would be through an interface and type reflection. On balance, I thought it was more understandable to enforce the type safety imposed by golang rather than bypass it with reflection. Keeping the golang type safety meant passing arguments of the right types, and all six maps are technically different types, even though they look similar. Happy to change if you can provide a construction that doesn't use reflection to ascertain the type, or a good rationale for why reflection is more understandable and sensible here.